### ISSUE TITLE
Wrong error message: "need nil, got nil"

### ISSUE THREAD

**Issue by ajeetdsouza** [elves/elvish#1387]
For this command:

```
$ builtin:show $nil
```

The output is:
```
Exception: wrong type of argument 0: wrong type: need nil, got nil
[tty 16], line 1: builtin:show $nil
```

`builtin:show` actually needs an object of type `exception`.

**Comment by krader1961** [comment]
Dup of issue #715 and #744. I have a fix out for review (PR #1360) that I need to find a few minutes to update.

**Comment by ajeetdsouza** [comment]
Thanks!
